src/plot_api/plot_api.js

```diff
@@ -2508,6 +2457,373 @@ Plotly.relayout = function relayout(gd, astr, val) {
+                    frameOpts: getFrameOpts(i),
+                    transitionOpts: getTransitionOpts(i)
+                };
+
+                if(i === frameList.length - 1) {
+                    nextFrame.onComplete = resolve;
+                    nextFrame.onInterrupt = reject;
+                }
+
+                trans._frameQueue.push(nextFrame);
+            }
+
+            if(!trans._animationRaf) {
+                beginAnimationLoop();
+            }
+        }
+
+        function stopAnimationLoop() {
+            cancelAnimationFrame(trans._animationRaf);
+            trans._animationRaf = null;
+        }
+
+        function nextFrame() {
+            if(trans._currentFrame) {
+                if(trans._currentFrame.onComplete) {
+                    trans._currentFrame.onComplete();
+                }
+            }
+
+            var newFrame = trans._currentFrame = trans._frameQueue.shift();
+
+            if(newFrame) {
+                trans._lastframeat = Date.now();
+                trans._timetonext = newFrame.frameOpts.duration;
+
+                Plots.transition(gd,
+                    newFrame.frame.data,
+                    newFrame.frame.layout,
+                    newFrame.frame.traces,
+                    newFrame.frameOpts,
+                    newFrame.transitionOpts
+                ).then(function() {
+                    if(trans._frameQueue.length === 0) {
+                        gd.emit('plotly_animated');
+                        if(trans._currentFrame && trans._currentFrame.onComplete) {
+                            trans._currentFrame.onComplete();
+                            trans._currentFrame = null;
+                        }
+                    }
+                });
+            }
+
+            if(trans._frameQueue.length === 0) {
+                stopAnimationLoop();
+                return;
+            }
+        }
+
+        function beginAnimationLoop() {
+            gd.emit('plotly_animating');
+
+            var canAnimateSynchronously = !trans._animationRaf && trans._frameQueue.length === 1;
+
+            if(canAnimateSynchronously) {
+                // If there is no animation running and only one frame has been received, then
+                // simply transition this frame synchonously and avoid starting and stopping the
+                // timing loop.
+                nextFrame();
+                return;
+            }
+
+            // If no timer is running, then set last frame = long ago:
+            trans._lastframeat = 0;
+            trans._timetonext = 0;
+            trans._runningTransitions = 0;
+            trans._currentFrame = null;
+
+            var doFrame = function() {
+                // Check if we need to pop a frame:
+                if(Date.now() - trans._lastframeat > trans._timetonext) {
+                    nextFrame();
+                }
+
+                trans._animationRaf = requestAnimationFrame(doFrame);
```

Comment:
Added a RAF polyfill. See: #904. If that looks good, then as part of that, the range slider example should be refactored to avoid the if/then.